src/main.c

```diff
@@ -272,7 +273,23 @@ main_loop(void)
 	/* TODO: set listening socket */
 	debug(LOG_NOTICE, "Created web server on %s", config->gw_address);
 
+	if (config->preauth) {
+		debug(LOG_NOTICE, "Preauth is Enabled - Overiding FAS configuration.\n");
+		debug(LOG_NOTICE, "Preauth Script is %s\n", config->preauth);
+
+		//override all other FAS settings
+		config->fas_remoteip = safe_strdup(config->gw_ip);
+		config->fas_remotefqdn = NULL;
+		config->fas_key = NULL;
+		config->fas_port = config->gw_port;
+		safe_asprintf(&preauth_dir, "/%s/", config->preauthdir);
+		config->fas_path = safe_strdup(preauth_dir);
```

Comment:
This would be shorter and needs to temporary.
```
safe_asprintf(&config->fas_path, "/%s/", config->preauthdir);
```

Reply:
Indeed.